### ISSUE TITLE
nsobject_prefer_isequal false positive in 0.50.0-rc2

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#4404]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

False positive with struct nested inside class.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.50.0-rc2
* Installation method used (Homebrew, CocoaPods, building from source, etc)? release page
* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.
No
* Which Xcode version are you using (check `xcodebuild -version`)?
14.0.1

* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
public final class A: NSObject {
    public enum B: Equatable {
        case c

        public func hash(into hasher: inout Hasher) {
            filename.original.hash(into: &hasher)
        }

        public static func == (lhs: B, rhs: B) -> Bool {
            return true
        }
    }
}
```

**Comment by jpsim** [comment]
Ah yes I see how this can happen. @marcelofabri did you want to fix this one? If not I can tackle it.

**Comment by marcelofabri** [comment]
I won't get to it right away, so feel free to grab it 🙏
